### PR TITLE
[JENKINS-65237] Bump Groovy from 2.4.12 to 2.4.21

### PR DESCRIPTION
**Pull request by basil**
Adapting to jenkinsci/jenkins#5939, specifically [this comment](https://github.com/jenkinsci/jenkins/pull/5939#pullrequestreview-818391176):

> [GROOVY-8428](https://issues.apache.org/jira/browse/GROOVY-8428) appears to be the only new APIs that could be relevant to Jenkins users. I think it would make sense to add them to `generic-whitelist` in `script-security` when `groovy-sandbox` is updated assuming the Jenkins baseline in the plugin is updated as well.

The change to `GroovyCallSiteSelectorTest` adapts a test that was using a Guava class that was removed in the latest version of Guava, which is now present in core. I resolved this issue by replacing the usage with the equivalent class from [Apache Commons IO](https://commons.apache.org/proper/commons-io/).

### CODE REVIEW

File: pom.xml
Comment (jglick) [inline pom.xml:20]:
```suggestion
    <jenkins.version>2.326</jenkins.version>
```

File: pom.xml
Comment (jglick) [inline pom.xml:55]:
```suggestion
        <artifactId>bom-weekly</artifactId>
```

**Review by jglick — APPROVED** [review]

**Comment by jglick** [comment]
Should probably first wait for #368 to be released, then merge #382, then this.

**Comment by basil** [comment]
Yeah, I'm not in any rush for this to be merged. I just wanted to file the PR to close the loop on remaining tasks regarding the Groovy upgrade.

**Comment by NotMyFault** [comment]
Heyo @basil,

do you look forward to merge this PR anytime soon? See my PR linked, it would depend on the baseline bump you look forward to do.

**Comment by basil** [comment]
> do you look forward to merge this PR anytime soon?

I am not a maintainer, but perhaps @car-roll could consider merging this now that we have an additional use case in mind.

File: pom.xml
Comment (timja) [inline pom.xml:54]: Weekly should only be used if you are actively tracking the latest LTS version.
Otherwise the closest LTS line should be used

```suggestion
        <artifactId>bom-2.319.x</artifactId>
```

**Review by car-roll — APPROVED** [review]
Will merge once @timja 's comment is addressed

File: pom.xml
Comment (jglick) [reply pom.xml:54]: Rather switch to https://github.com/jenkinsci/bom/pull/876 I guess.

Comment (timja) [reply pom.xml:54]: 319 should be fine for now 😄 

can be bumped to next LTS separately (also fine to bump to next LTS but need to get the `ldap` plugin tests sorted first).

Comment (jglick) [reply pom.xml:54]: https://github.com/jenkinsci/bom/releases/tag/1155.v77b_fd92a_26fc is ready.

Comment (timja) [reply pom.xml:54]: @car-roll I would suggest as the maintainer just commit whatever you feel like here 😄

**Review by timja — APPROVED** [review]

**Pull request merged**
